backend/Controllers/UserController.cs

```diff
@@ -21,4 +23,59 @@ public async Task<IActionResult> GetByUserName([FromRoute] string userName)
         }
         return Ok(new UserResponse { Id = matchingUser.Id, UserName = matchingUser.UserName!, });
     }
+
+    [HttpGet("GetUserId/{userId}")]
+    public async Task<User?> GetByUserId([FromRoute] string userId)
+    {
+        var matchingUser = await _userManager.FindByIdAsync(userId);
+        if (matchingUser == null)
+        {
+            return null;
+        }
+        return matchingUser;
+    }
+
+    [HttpPost("UpdateUser")]
+    public async Task<IActionResult> UpdateUser(UpdateUserRequest userRequest)
+    {
+        User? user = await GetByUserId(userRequest.Id.ToString());
+        var errorResponse = new ErrorResponse();
+        var generalErrors = new List<string>();
+
+        if (user == null)
+        {
+            return NotFound();
+        }
+
+        if (user.IsSuspended)
+        {
+            errorResponse.Errors["User is suspended"] = generalErrors;
+            return BadRequest(errorResponse);
+        }
+
+        if (userRequest.FirstName != null)
+        {
+            user.FirstName = userRequest.FirstName;
+        }
+        if (userRequest.LastName != null)
+        {
+            user.LastName = userRequest.LastName;
+        }
+        if (userRequest.AboutMe != null)
+        {
+            user.AboutMe = userRequest.AboutMe;
+        }
+
+        _context.Users.Update(user);
+        await _context.SaveChangesAsync();
```

Comment:
The database interactions shouldn't be happening in the controller - they should be in the User Service